### PR TITLE
vmm: Add support for notifying guest is being paused

### PR DESCRIPTION
**Pull request by sboeuf**
Extend the Vcpu trait to be able to notify a guest that it is being paused. This is useful to avoid watchdogs running in the guest to complain when the guest is resumed after a certain period of time.

### CODE REVIEW

File: vmm/src/cpu.rs [hunk 1]
```diff
@@ -1368,6 +1382,10 @@ impl Snapshottable for CpuManager {
             let vcpu = self
                 .create_vcpu(cpu_id.parse::<u8>().unwrap(), None, Some(*snapshot.clone()))
                 .map_err(|e| MigratableError::Restore(anyhow!("Could not create vCPU {:?}", e)))?;
+
+            // Adding vCPU to the CpuManager's vCPU list.
+            self.vcpus.push(vcpu.clone());
+
```
Comment (sameo) [inline vmm/src/cpu.rs:1388]: How is this change related to the guest notification?

Comment (sboeuf) [reply vmm/src/cpu.rs:1388]: Hehe it's not, that's why I'm gonna push it in its own commit :)

Comment (sboeuf) [reply vmm/src/cpu.rs:1388]: done

File: vmm/src/cpu.rs [hunk 2]
```diff
@@ -1368,6 +1368,10 @@ impl Snapshottable for CpuManager {
...
```
Comment (sameo) [inline vmm/src/cpu.rs:1388]: Shouldn't this be part of create_vcpu instead?

Comment (sboeuf) [reply vmm/src/cpu.rs:1388]: We can, but if we do so, we have to modify the "boot from scratch" codepath.
